net.go

```diff
@@ -126,41 +170,59 @@ func dialSession(d net.Dialer, network, addr string, ctx *Ctx, flags DialFlags,
 		}
 		// TODO: use operating system default certificate chain?
 	}
+	return ctx, nil
+}
 
-	c, err := d.Dial(network, addr)
+func createConnection(context context.Context, d net.Dialer, network,
+	addr string) (net.Conn, error) {
+	c, err := d.DialContext(context, network, addr)
```

Comment:
It could be simplified and unified in such way:
```suggestion
func createConnection(context context.Context, network, addr string) (net.Conn, error) {
	return net.Dialer{}.DialContext(context, network, addr)
```
and usage in the timeout case (`DialTimeout`) will be:
```
ctx, cancel := context.WithTimeout(context.Background(), timeout)
defer cancel()
conn, err := createConnection(ctx, network, addr)
```

Reply:
Updated. I think we can keep `createConnection` function, because now I use it for parsing `host` too.